hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/common/HoodieFlinkEngineContext.java

```diff
@@ -236,7 +236,7 @@ public <K extends Comparable<K>, V extends Comparable<V>, R> HoodieData<R> mapGr
                                                                                             SerializableFunction<Iterator<V>, Iterator<R>> processFunc,
                                                                                             List<K> keySpace,
                                                                                             boolean preservesPartitioning) {
-    throw new UnsupportedOperationException("processKeyGroups() is not supported in FlinkEngineContext");
+    return super.mapGroupsByKey(data, processFunc, keySpace, preservesPartitioning);
```

Comment:
Do you think we need parallelizing the process at key level, that would be a tradeoff between of resources and performance. cc @danny0405 @xushiyan

Reply:
I think we can paralle it first if Spark also did so, and come back to a fix when OOM is an issue there.

Reply:
+1

Reply:
will update.